app/src/layout.py

```diff
@@ -23,11 +24,25 @@
     plotter = plotter_factory.create_plotter(data[plot_key], config[plot_key])
     plotter.generate()
 
+    location_toggle = Select(options={
+        'Deutschland': 'de', 'Berlin': 'ber'}, value='de')
+
+    def create_update_chart(flex_obj, plotter):
+        def update_chart(event):
+            selected_location = event.new
+            flex_obj.__setitem__(1, plotter.plots[selected_location])
+        return update_chart
+
     # Place plots and filters in the layout
-    flex_obj = FlexBox(*plotter.plots.values(),
+    flex_obj = FlexBox(location_toggle,
+                       #    *plotter.plots.values(),
+                       plotter.plots[location_toggle.value],
                        *plotter.filters.values(),
                        flex_direction="column",
                        align_items="center",
                        sizing_mode="stretch_width")
 
+    update_chart = create_update_chart(flex_obj, plotter)
```

Comment:
I think we can simplify the code a lot by deleting this line and replacing the `create_update_chart` function with:

```
def update_chart(self, event):
    flex_obj.__setitem__(1, plotter.plots[event.new])
```

(I didn't test it, lmk if it doesn't work and we can figure it out together)

Reply:
Hmm. I didn't get it running and from how I understand it, the factory function is needed, because we have multiple flexboxes to update. but we can look into it again.

Reply:
You are right! But it can still be simplified a bit, I commited a couple small changes.

- As the function inside `create_update_chart` is only used once inside of it, we can make use python's lambda (an anonymous function with a simpler one-line syntax)
- This function should be defined outside the loop, otherwise we define it over and over with each loop, which is unnecessary